lib_eio_luv/eio_luv.ml

```diff
@@ -126,6 +126,26 @@ let await_with_cancel ~request fn =
         )
     )
 
+let await_with_cancel_poll ~poll fn =
+  enter (fun st k ->
+    let cancel_reason = ref None in
+    Eio.Private.Fibre_context.set_cancel_fn k.fibre (fun ex ->
+        cancel_reason := Some ex;
+        match Luv.Poll.stop poll with
+        | Ok () -> ()
```

Comment:
You need to make sure that the continuation is called in this case. For files, `Request.cancel` results in the callback being called with a cancelled result. I suspect this is not the case for `Poll.stop`.